### PR TITLE
feat: add support for strategic merge patches

### LINKED ISSUE #1259: feat: add support for strategic merge patches
Currently, Sidero supports only **RFC 6902** patches (aka JSON patch). [1]

The `talosctl` tool supports also the **strategic merge** patches. [2]
This patching style is used in customisation approaches which aim to preserve the config structure, e.g., for clearer understanding, complex structure of patches.

Additionally, the strategic merge patch in `talosctl` supports the newly-introduced multiple-document configuration unlike the RFC 6902 approach. [3]

This feature in Sidero depends on the analogous feature in CABPT. [4]

[1] https://www.sidero.dev/v0.6/guides/patching/
[2] https://www.talos.dev/v1.6/talos-guides/configuration/patching/#strategic-merge-patches
[3] https://www.talos.dev/v1.6/talos-guides/configuration/patching/#configuration-patch-formats
[4] https://github.com/siderolabs/cluster-api-bootstrap-provider-talos/issues/181

### PR DESCRIPTION
**Pull request by ksawio97**
This commit implements strategic merge patch by reusing Talos configpatcher patch handling. Furthermore, a unit test scenario has been added to ensure its functionality.

Resolves #1259.

### CODE REVIEW

**Review by yoctozepto — APPROVED** [review]
LGTM, some nitpicks in comments and messages only

File: app/sidero-controller-manager/internal/metadata/fixture_test.go [hunk 1]
```diff
@@ -217,6 +218,80 @@ cluster: {}
 `)
 }
 
+// fixture6 creates a server with Server- & ServerClass-level with strategic merge config patches.
```
[the remark this answers is not in the document]
Comment (ksawio97) [reply app/sidero-controller-manager/internal/metadata/fixture_test.go:221]: It's based on fixture3, but rfc6902 config patches are replaced with merge config patches so I think this comment's appropriate.

[the remark this answers is not in the document]
Comment (yoctozepto) [reply app/sidero-controller-manager/internal/metadata/fixture_test.go:221]: Yes, you are right.

**Review by yoctozepto — APPROVED** [review]
LGTM, thank you

**Review by smira — APPROVED** [review]
thank you!

**Review by frezbo — APPROVED** [review]

**Pull request merged**